pkg/codegen/schema/schema.go

```diff
@@ -431,7 +442,7 @@ type Resource struct {
 // }
 //
 // A.ReplaceOnChanges() == [[B, D], [C]]
-func (r *Resource) ReplaceOnChanges() (changes [][]*Property, err []error) {
+func (r *Resource) ReplaceOnChanges() (changes [][]*Property, overrides []string, err []error) {
```

Comment:
Thanks for the clarification there, good point. I will give it a shot and report back.

Reply:
> Would it be possible to fold overrides into changes? changes is a list of property paths where replaceOnChanges should be emitted. This caries the same meaning as overrides. 

Looking into this a bit more, I don't believe `changes [][]*Properties` is sufficient to encapsulate the patterns in overrides. For instance, how would we encode `*` or `a.b[1]` or `a.b["key"]`? I understand some of the latter examples are a bit contrived but are valid patterns to specify to `replaceOnChanges`. I worry this might turn into a yak-shave to fit free-form string pattern to typed data structure - only to be re-translated into a free form string pattern at SDK generation point. I feel validating the patterns is a much more constrained problem to solve if we leave `overrides` as separate? Granted it comes at the cost of elegance.

Reply:
That makes a lot of sense. I thought about how to accommodate `*` but I forgot `a.b[1]` and `a.b["key"]`. Your right, and I do want to avoid the yak-shaving. 

I still like the idea of abstracting over both `overrides` and `changes`. I really think that can be encapsulated. It might be smarter to move the other direction, and return only a list of validated `PropertyPath`s. I would prefer that we return a non-string data structure since some languages (python) will need to parse the string later. `PropertyPath` should have full fidelity, since it is what override strings get parsed back into at runtime.

What do you think?

Reply:
I agree, will try this out!